src/groovy/au/org/emii/portal/wms/GeoserverServer.groovy

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2014 IMOS
+ *
+ * The AODN/IMOS Portal is distributed under the terms of the GNU General Public License
+ *
+ */
+
+package au.org.emii.portal.wms
+
+import au.org.emii.portal.Layer
+import au.org.emii.portal.Server
+import au.org.emii.portal.proxying.ExternalRequest
+import groovy.xml.MarkupBuilder
+
+class GeoserverServer extends WmsServer {
```

Comment:
Possibly the filter specific stuff can be extracted in to a separate class dealing with just that.  Then this class becomes a pretty high level interface and it's easier to see what can be done with it (without the low-level stuff providing a distraction).